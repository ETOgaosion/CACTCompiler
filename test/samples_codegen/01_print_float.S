iobuff:
	.zero 80
	.equ	ImmValue_0, 0x40133333

.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	li	t0, ImmValue_0
	fmv.w.x	ft0, t0
	fsw	ft0, -12(sp)
	flw	fa0, -12(sp)
	addi	sp, sp, -16
	call	print_float
	addi	sp, sp, 16
	mv	a0, zero
	ld	ra, -8(sp)
	ret
