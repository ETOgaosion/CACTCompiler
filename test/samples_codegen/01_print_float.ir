============ global var ================
============ imm value ================
ImmValue_0: 2.3f
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
a := symbol type: 1; data type: 3; initValue: ImmValue_0: 2.3f

----------- tempVars ------------
----------- Codes ------------
main:
	a = ImmValue_0: 2.3f;
	addparam a;
	call print_float, frame size: 16; 
	return 0;

