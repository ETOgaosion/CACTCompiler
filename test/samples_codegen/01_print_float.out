bbl loader
2.300000
